### PR TITLE
fix#1513  Add constr for DataContext(providerName, connectionString)

### PR DESCRIPTION
**Pull request by GonzoSpire**
and Tests

### CODE REVIEW

File: Source/LinqToDB/DataContext.cs [hunk 1]
```diff
@@ -57,6 +57,24 @@ public DataContext([JetBrains.Annotations.NotNull] IDataProvider dataProvider, [
...
+		/// <summary>
+		/// Creates data context using specified database provider and connection string.
+		/// </summary>
+		/// <param name="providerName">Name of database provider to use with this connection. <see cref="ProviderName"/> class for list of providers.</param>
+		/// <param name="connectionString">Database connection string to use for connection with database.</param>
+		public DataContext(
+			[JetBrains.Annotations.NotNull] string providerName,
+			[JetBrains.Annotations.NotNull] string connectionString)
+		{
+			if (providerName == null) throw new ArgumentNullException(nameof(providerName));
```
Comment (sdanyliv) [inline Source/LinqToDB/DataContext.cs]: Follow table formatting rules.

File: Tests/Linq/Linq/DataContextTests.cs [hunk 2]
```diff
@@ -71,5 +71,23 @@ public class DataContextTests : TestBase
...
+
+		[Test]
+		public void issue1513Test([DataSources(false)]string context)
+		{
+			using (var db = (TestDataConnection)GetDataContext(context))
+			{
+				var _db = new DataContext(db.DataProvider.Name, db.ConnectionString);
+
+				Assert.AreEqual(db.DataProvider.Name, _db.DataProvider.Name);
+				Assert.AreEqual(db.ConnectionString, _db.ConnectionString);
```
Comment (sdanyliv) [inline Tests/Linq/Linq/DataContextTests.cs]: Better to check fails here by putting wrong provider name and correct connection string, or wrong connection string with real provider name.

Comment (GonzoSpire) [reply Tests/Linq/Linq/DataContextTests.cs]: wrong connection string does not cause exceptions, and should?

Comment (sdanyliv) [reply Tests/Linq/Linq/DataContextTests.cs]: Just select first record of Child

**Pull request merged**